### PR TITLE
Angular, React, and Vue sample.

### PR DESCRIPTION
**Pull request by IEvangelist**
A sample that shows how to use .NET Aspire with Angular, React, and Vue.

**.NET Aspire dashboard**

image

**Angular app**

image

**React app**

image

**Vue app**

image

### CODE REVIEW

**Comment by OwenCross289** [comment]
When checking out this commit, I followed the readme, ran npm install for all 3 of the JS projects. When running the Aspire AppHost project, the Angular app fails to start: it gets a NaN as a port, with this logged.

```
> angular-weather@0.0.0 start
> ng serve --port %PORT%
Node.js version v21.1.0 detected.
Odd numbered Node.js versions will not enter LTS status and should not be used for production. For more information, please see https://nodejs.org/en/about/previous-releases/.
/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/rxjs/dist/cjs/internal/util/reportUnhandledError.js:13
            throw err;
            ^
RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received type number (NaN).
    at validatePort (node:internal/validators:424:11)
    at Server.listen (node:net:2000:5)
    at /Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/utils/check-port.js:55:14
    at new Promise (<anonymous>)
    at checkPort (/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/utils/check-port.js:27:12)
    at initialize (/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/builders/dev-server/builder.js:112:63) {
  code: 'ERR_SOCKET_BAD_PORT'
}
Node.js v21.1.0
```
Set up:
Aspire preview 2 workload installed.
Running through dotnet run in AppHost proj
MacOs: 14.2.1
Node: v21.1.0
npm ng --version: 10.2.3

Sorry if I am being too early on this

**Comment by IEvangelist** [comment]
> When checking out this commit, I followed the readme, ran npm install for all 3 of the JS projects. When running the Aspire AppHost project, the Angular app fails to start: it gets a NaN as a port, with this logged.
> 
> ```
> > angular-weather@0.0.0 start
> > ng serve --port %PORT%
> Node.js version v21.1.0 detected.
> Odd numbered Node.js versions will not enter LTS status and should not be used for production. For more information, please see https://nodejs.org/en/about/previous-releases/.
> /Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/rxjs/dist/cjs/internal/util/reportUnhandledError.js:13
>             throw err;
>             ^
> RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received type number (NaN).
>     at validatePort (node:internal/validators:424:11)
>     at Server.listen (node:net:2000:5)
>     at /Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/utils/check-port.js:55:14
>     at new Promise (<anonymous>)
>     at checkPort (/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/utils/check-port.js:27:12)
>     at initialize (/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/builders/dev-server/builder.js:112:63) {
>   code: 'ERR_SOCKET_BAD_PORT'
> }
> Node.js v21.1.0
> ```
> 
> Set up: Aspire preview 2 workload installed. Running through dotnet run in AppHost proj MacOs: 14.2.1 Node: v21.1.0 npm ng --version: 10.2.3
> 
> Sorry if I am being too early on this

Ah, I think that might be an issue with Windows. I passed the port to the `ng serve --port` switch using `%PORT%`, if you're not on Windows that won't work... Try changing that to `ng serve --port $PORT` instead.

**Comment by OwenCross289** [comment]
> > When checking out this commit, I followed the readme, ran npm install for all 3 of the JS projects. When running the Aspire AppHost project, the Angular app fails to start: it gets a NaN as a port, with this logged.
> 
> > 
> 
> > ```
> 
> > > angular-weather@0.0.0 start
> 
> > > ng serve --port %PORT%
> 
> > Node.js version v21.1.0 detected.
> 
> > Odd numbered Node.js versions will not enter LTS status and should not be used for production. For more information, please see https://nodejs.org/en/about/previous-releases/.
> 
> > /Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/rxjs/dist/cjs/internal/util/reportUnhandledError.js:13
> 
> >             throw err;
> 
> >             ^
> 
> > RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received type number (NaN).
> 
> >     at validatePort (node:internal/validators:424:11)
> 
> >     at Server.listen (node:net:2000:5)
> 
> >     at /Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/utils/check-port.js:55:14
> 
> >     at new Promise (<anonymous>)
> 
> >     at checkPort (/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/utils/check-port.js:27:12)
> 
> >     at initialize (/Users/owen/Developer/aspire-samples/samples/AspireWithJavaScript/AspireJavaScript.Angular/node_modules/@angular-devkit/build-angular/src/builders/dev-server/builder.js:112:63) {
> 
> >   code: 'ERR_SOCKET_BAD_PORT'
> 
> > }
> 
> > Node.js v21.1.0
> 
> > ```
> 
> > 
> 
> > Set up: Aspire preview 2 workload installed. Running through dotnet run in AppHost proj MacOs: 14.2.1 Node: v21.1.0 npm ng --version: 10.2.3
> 
> > 
> 
> > Sorry if I am being too early on this
> 
> 
> 
> Ah, I think that might be an issue with Windows. I passed the port to the `ng serve --port` switch using `%PORT%`, if you're not on Windows that won't work... Try changing that to `ng serve --port $PORT` instead.

That, fixed it! Thanks!

**Comment by IEvangelist** [comment]
Note to self, update this to use an `.env` with `DEV_SERVER_PORT=PORT` or change `PORT` to `DEV_SERVER_PORT`.

File: samples/AspireWithJavaScript/AspireJavaScript.AppHost/Program.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+var builder = DistributedApplication.CreateBuilder(args);
+
+var weatherApi =
+    builder.AddProject<Projects.AspireJavaScript_MinimalApi>("weatherApi");
```
Comment (OwenCross289) [inline samples/AspireWithJavaScript/AspireJavaScript.AppHost/Program.cs:4]: I don't know if these samples are suppose to work with the 'azd init' & 'azd up' but Azure does not support uppercase chars in the names. So to get a deployment I had to change this to `var weatherApi = builder.AddProject<Projects.AspireJavaScript_MinimalApi>("weatherapi");` and propigate it through.

File: samples/AspireWithJavaScript/AspireJavaScript.AppHost/Program.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+// Angular: npm run start
+builder.AddNpmApp("angular", "../AspireJavaScript.Angular")
+    .WithReference(weatherApi)
+    .WithServiceBinding(scheme: "http", env: "PORT")
```
Comment (OwenCross289) [inline samples/AspireWithJavaScript/AspireJavaScript.AppHost/Program.cs:9]: Also to enable the azd, the service bindings each need to specify a port. I just copied the node sample and did '.WithServiceBinding(containerPort: 3000, scheme: "http", env: "PORT")' and incremented the port for React and Vue.

Both of these changes got them to deploy with azd but I still could not get the public facing deployed websites to load.

Comment (IEvangelist) [reply samples/AspireWithJavaScript/AspireJavaScript.AppHost/Program.cs:9]: Awesome, thanks for catching this too!

**Review by mitchdenny — APPROVED** [review]
LGTM. I do wonder whether we could make these projects show up in VS via the esproj project types.

**Comment by IEvangelist** [comment]
> LGTM. I do wonder whether we could make these projects show up in VS via the esproj project types.

Yeah, that would be really cool!

**Pull request merged**